opentelemetry-api/src/opentelemetry/util/_once.py

```diff
@@ -0,0 +1,47 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from threading import Lock
+from typing import Callable
+
+
+class Once:
```

Comment:
I'm thinking to put this in a PR to main branch so we can use it for both tracing and metrics without waiting for metrics to merge.

Reply:
Created https://github.com/open-telemetry/opentelemetry-python/pull/2182. We can merge that into `metrics_new` and this will go away